Sources/ResilientDecoding/Resilient.swift

```diff
@@ -65,15 +65,35 @@ public struct Resilient<Value: Decodable>: Decodable {
 // MARK: - Decoding Outcome
 
 #if DEBUG
+/**
+ The outcome of decoding a `Resilient` type
+ */
 public enum ResilientDecodingOutcome {
+  /**
+   A value was decoded successfully
+   */
   case decodedSuccessfully
+  
+  /**
+   The key was missing, and it was not treated as an error (for instance when decoding an `Optional`)
+   */
   case keyNotFound
+  
+  /**
+   The value was `nil`, and it was not treated as an error (for instance when decoding an `Optional`)
+   */
   case valueWasNil
   
-  /// https://github.com/apple/swift/blob/88b093e9d77d6201935a2c2fb13f27d961836777/stdlib/public/Darwin/Foundation/JSONEncoder.swift#L1657-L1661
+  /**
+   An error was recovered from during decoding
+   - parameter `wasReported`: Some errors are not reported, for instance `ArrayDecodingError`
```

Comment:
Can you help me understand why `ArrayDecodingError` is not reported? Is it because it's a "meta error" or something? I didn't fully follow that part.

Reply:
Yes, this is in comments around `ArrayDecodingError`. The basic idea is `ArrayDecodingError` actually represents a number of errors that happened at _indices_ of the array being decoded. Reporting that error would effectively be reporting the index-level errors twice.